cmpserver/plugin/plugin.go

```diff
@@ -82,6 +103,9 @@ func (s *Service) GenerateManifest(ctx context.Context, q *apiclient.ManifestReq
 
 // MatchRepository checks whether the application repository type is supported by config management plugin server
 func (s *Service) MatchRepository(ctx context.Context, q *apiclient.RepositoryRequest) (*apiclient.RepositoryResponse, error) {
+	bufferedCtx, cancel := buffered_context.WithEarlierDeadline(ctx, 100*time.Millisecond)
```

Comment:
I think this time is internal implementation details, so no need to configure it. I would introduce a const though.